### PR TITLE
Add an additional resource to DI topic

### LINKED ISSUE #8641: Forwarded-types (registering multiple interfaces for a service)
@Rick-Anderson - a recent post on registering forwarded types - [How to register a service with multiple interfaces in ASP.NET Core DI](https://andrewlock.net/how-to-register-a-service-with-multiple-interfaces-for-in-asp-net-core-di/)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0d78f540-81fd-1ba9-cb90-591e4ec27d9b
* Version Independent ID: 2251f557-dd3a-ec35-2558-6243337db009
* Content: [Dependency injection in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-2.1#design-services-for-dependency-injection)
* Content Source: [aspnetcore/fundamentals/dependency-injection.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/dependency-injection.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8641 

... also @pakrym, can you say if this would have wide enough interest to become its own topic on the doc set?

### CODE REVIEW

File: aspnetcore/fundamentals/dependency-injection.md [hunk 1]
```diff
@@ -560,3 +560,4 @@ Dependency injection is an *alternative* to static/global object access patterns
 * [Explicit Dependencies Principle](https://deviq.com/explicit-dependencies-principle/)
 * [Inversion of Control Containers and the Dependency Injection Pattern (Martin Fowler)](https://www.martinfowler.com/articles/injection.html)
 * [New is Glue ("gluing" code to a particular implementation)](https://ardalis.com/new-is-glue)
+* [How to register a service with multiple interfaces in ASP.NET Core DI](https://andrewlock.net/how-to-register-a-service-with-multiple-interfaces-for-in-asp-net-core-di/)
```
Comment (scottaddie) [inline aspnetcore/fundamentals/dependency-injection.md]: How to register --> Register

File: aspnetcore/fundamentals/dependency-injection.md [hunk 2]
```diff
@@ -560,3 +560,4 @@ Dependency injection is an *alternative* to static/global object access patterns
 * [Explicit Dependencies Principle](https://deviq.com/explicit-dependencies-principle/)
 * [Inversion of Control Containers and the Dependency Injection Pattern (Martin Fowler)](https://www.martinfowler.com/articles/injection.html)
```
Comment (scottaddie) [inline aspnetcore/fundamentals/dependency-injection.md]: Dependency Injection --> DI
- this suggestion should be applied all throughout the doc (perhaps in a separate PR)

Comment (guardrex) [reply aspnetcore/fundamentals/dependency-injection.md]: The title of the linked content is ...

> Inversion of Control Containers and the Dependency Injection pattern

Comment (guardrex) [reply aspnetcore/fundamentals/dependency-injection.md]: > perhaps in a separate PR

Yes, this wasn't a UE pass. The goal of the PR is just to resolve the issue.

File: aspnetcore/fundamentals/dependency-injection.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/dependency-injection.md]: Same thing ... the title is ...

> How to register a service with multiple interfaces in ASP.NET Core DI

Comment (guardrex) [reply aspnetcore/fundamentals/dependency-injection.md]: I need guidance (cc: @Rick-Anderson). I've been titling 3rd party *Additional resources* links by what the author of the content used as the title. Should I start changing the titles displayed in the docs in this section to conform to our titling rules? I don't recommend it, but I certainly will follow whatever rule is laid down.

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/dependency-injection.md]: What's wrong with `## Additional resources` ?

Comment (guardrex) [reply aspnetcore/fundamentals/dependency-injection.md]: The question is should we use the titles exactly from pages that we link to as our link text or should we make their titles conform to our title guidelines in our link text?

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/dependency-injection.md]: I wouldn't force them to conform to our style, but I sometimes change them up if they read better. It's rare we link to 3rd parties.

Comment (guardrex) [reply aspnetcore/fundamentals/dependency-injection.md]: Yeah ... I've been leaving them as the author has them. I would change a title in our link text if there was a spelling error or something along those lines. Thus far, 3rd party titles have been good.

**Pull request merged**